mlops/london_taxi/src/mlops_pipeline.py

```diff
@@ -231,16 +232,23 @@ def execute_pipeline(
                     ):
                         break
                 else:
+                    print(
+                        f"Job {pipeline_job.name} exceeded the wait time limit of 1 hour."
+                    )
                     break
 
             if pipeline_job.status == "Completed" or pipeline_job.status == "Finished":
-                print("job completed")
+                print("Job completed successfully.")
             else:
-                raise Exception("Sorry, exiting job with failure..")
+                raise Exception(
+                    f"Job {pipeline_job.name} did not complete successfully. "
+                    f"Current status: {pipeline_job.status}"
+                )
     except Exception as ex:
```

Comment:
If possible, it would be nice if we could catch Authorization Errors separately so it's easier to know if it's a permission issue or something else.. Something like this:

except AuthorizationException as auth_ex:
     do something
except Exception as ex:
     do something
     
Note: I don't know what the AuthorizationException actually would be. ChatGPT suggest `from azure.core.exceptions import ClientAuthenticationError` so maybe something like that.

Reply:
Changed this to catching separately as suggested